q2_feature_classifier/_consensus_assignment.py

```diff
@@ -0,0 +1,243 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2017, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import tempfile
+import subprocess
+import pandas as pd
+from os.path import isfile
+from collections import Counter
+
+
+def _get_default_unassignable_label():
+    return "Unassigned"
+
+
+def _consensus_assignments(
+        cmd, ref_taxa, min_consensus=0.51, output_no_hits=False,
+        exp_seq_ids=None,
+        unassignable_label=_get_default_unassignable_label()):
+    '''Run command line subprocess and find consensus taxonomy.'''
+    with tempfile.NamedTemporaryFile() as output:
+        cmd = cmd + [output.name]
+        _run_command(cmd)
+        obs_taxa = _import_blast_assignments(
+            output.name, ref_taxa, unassignable_label=unassignable_label)
+        consensus = _compute_consensus_annotations(
+            obs_taxa, min_consensus=min_consensus,
+            unassignable_label=unassignable_label)
+        # If output_no_hits=True, find/add query IDs missing from consensus
+        if output_no_hits is True:
+            consensus = _output_no_hits(
+                consensus, exp_seq_ids, unassignable_label)
+        # if there are no consensus results (no hits for any query), default
+        # to empty result file (must create empty dict or fail on pd error)
+        if not consensus:
+            consensus = {'': ('', '')}
+        result = pd.DataFrame.from_dict(consensus, 'index')
+        result.index.name = 'Feature ID'
+        result.columns = ['Taxon', 'Confidence']
+        return result
+
+
+def _validate_params(min_id, maxaccepts, min_consensus):
```

Comment:
You can drop this function now, as well as all related tests. The `Range` parameters that I'm having you add will take care of this validation for you.